brokerapi/brokers/bigtable/broker.go

```diff
@@ -48,81 +46,50 @@ var storageTypes = map[string]googlebigtable.StorageType{
 
 // Provision creates a new Bigtable instance from the settings in the user-provided details and service plan.
 func (b *BigTableBroker) Provision(ctx context.Context, instanceId string, details brokerapi.ProvisionDetails, plan models.ServicePlan) (models.ServiceInstanceDetails, error) {
-	var err error
-	var params map[string]string
-
-	if len(details.RawParameters) == 0 {
-		params = map[string]string{}
-	} else if err = json.Unmarshal(details.RawParameters, &params); err != nil {
-		return models.ServiceInstanceDetails{}, fmt.Errorf("Error unmarshalling parameters: %s", err)
-	}
-
-	// Ensure there is a name for this instance
-	if _, ok := params["name"]; !ok {
-		params["name"] = name_generator.Basic.InstanceNameWithSeparator("-")
-	}
-
-	service, err := b.createClient(ctx)
+	provisionContext, err := serviceDefinition().ProvisionVariables(instanceId, details, plan)
 	if err != nil {
 		return models.ServiceInstanceDetails{}, err
 	}
 
-	var clusterId string
-	if len(params["name"]) > 20 {
-		clusterId = params["name"][:20] + "-cluster"
-	} else {
-		clusterId = params["name"] + "-cluster"
-	}
-	if userClusterId, clusterIdOk := params["cluster_id"]; clusterIdOk {
-		clusterId = userClusterId
-	}
-
-	numNodes, err := strconv.Atoi(plan.ServiceProperties["num_nodes"])
-	if err != nil {
-		return models.ServiceInstanceDetails{}, fmt.Errorf("Error converting num_nodes to int: %s", err)
-	}
+	instanceName := provisionContext.GetString("name")
```

Comment:
Does instanceName need to be checked for an empty value before being used?

Reply:
It will be once #303 is done, and the API client will catch it, but I'll add a friendly error in the meantime.